tsl/test/sql/continuous_aggs.sql

```diff
@@ -930,3 +930,90 @@ GROUP BY
 CALL refresh_continuous_aggregate('mat_types',NULL,NULL);
 SELECT * FROM mat_types;
 
+-------------------------------------------------------------------------------------
+-- Test issue #2616 where cagg view contains an experssion with several aggregates in
+
+CREATE TABLE water_consumption
+(
+    sensor_id   integer      NOT NULL,
+    timestamp   timestamp(0) NOT NULL,
+    water_index integer
+);
+
+SELECT create_hypertable('water_consumption', 'timestamp', 'sensor_id', 2);
+
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:42:30', 1030);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:42:40', 1032);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:42:50', 1035);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:43:30', 1040);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:43:40', 1045);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:43:50', 1050);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:44:30', 1052);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:44:40', 1057);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:44:50', 1060);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:45:30', 1063);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:45:40', 1067);
+INSERT INTO public.water_consumption (sensor_id, timestamp, water_index) VALUES (1, '2010-11-03 09:45:50', 1070);
+
+-- The test with the view originally reported in the issue.
+CREATE MATERIALIZED VIEW water_consumption_aggregation_minute
+            WITH (timescaledb.continuous, timescaledb.materialized_only = TRUE)
+AS
+SELECT sensor_id,
+       time_bucket(INTERVAL '1 minute', timestamp) + '1 minute' AS timestamp,
+       (max(water_index) - min(water_index))                    AS water_consumption
+FROM water_consumption
+GROUP BY sensor_id, time_bucket(INTERVAL '1 minute', timestamp)
+WITH NO DATA;
+
+CALL refresh_continuous_aggregate('water_consumption_aggregation_minute', NULL, NULL);
+
+-- The results of the view and the query over hypertable should be the same
+SELECT * FROM water_consumption_aggregation_minute;
```

Comment:
I would suggest adding ORDER BY here on both view and raw query here to avoid flaky behavior. I guess also applies below.

Reply:
Good point! Fixed.